src/components/pages/home/transaction-history.tsx

```diff
@@ -1,71 +1,122 @@
 "use client";
+import React, { useEffect, useState } from "react";
 import {
   LineChart,
   Line,
   XAxis,
   YAxis,
   Tooltip,
   ResponsiveContainer,
-  TooltipProps,
 } from "recharts";
 import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { ChartContainer, type ChartConfig } from "@/components/ui/chart";
 import { formatPrice } from "@/lib/utils";
 import { subDays, format, parseISO } from "date-fns";
 
-const data = Array.from({ length: 15 }, (_, i) => i).map((i) => {
-  const date = subDays(new Date(), 15 - i);
-  return {
-    name: format(date, "MMM d"),
-    date: format(date, "yyyy-MM-dd"),
-    transactionsCount: Math.floor(Math.random() * 200) + 400,
-    price: 2,
-  };
-});
+const chartConfig = {
+  desktop: {
+    label: "Desktop",
+    color: "#2563eb",
+  },
+  mobile: {
+    label: "Mobile",
+    color: "#60a5fa",
+  },
+} satisfies ChartConfig;
 
-const CustomTooltip = ({ active, payload }: TooltipProps<number, string>) => {
+type DataPoint = {
+  name: string;
+  date: string;
+  price: number;
+};
+
+const CustomTooltip = ({ active, payload }: any) => {
```

Comment:
Why removing typing here? We don't want any, please reintroduce `TooltipProps<number, string>`.

Reply:
fixed